OpenRA.Mods.Common/Traits/Demolishable.cs

```diff
@@ -38,14 +44,34 @@ public bool PreventsEjectOnDeath(Actor self)
 			return info.PreventsEjectOnDeath;
 		}
 
-		public void Demolish(Actor self, Actor saboteur)
+		public void Demolish(Actor self, Actor saboteur, int delay)
 		{
-			self.Kill(saboteur);
+			if (this.delay > 0)
+				return;
+
+			this.delay = delay;
+			this.saboteur = saboteur;
+
+			modifiers = self.TraitsImplementing<IDamageModifier>()
```

Comment:
This works, but only because it is lazily evaluated by the `ApplyPercentageModifiers` below.  If somebody (e.g. the save/load code) decided to `.ToArray()` it then it would regress being able to use the iron curtain to cancel a demolish.